make.js

```diff
@@ -66,37 +69,74 @@ target.build = function() {
-        // Common: build, copy, install 
-        //--------------------------------
-        if (makeOptions.hasOwnProperty('common')) {
-            var common = makeOptions['common'];
-
-            common.forEach(function(mod) {
-                var commonPath = path.join(taskPath, mod['module']);
-                var modName = path.basename(commonPath);
-                var modOutDir = path.join(commonOutputPath, modName);
-
-                if (!test('-d', modOutDir)) {
-                    if (mod.type === 'node' && mod.compile == true) {
-                        buildNodeTask(commonPath, modOutDir);
-                        copyTaskResources(commonPath, modOutDir);
-
-                        // install the common module to the task dir
-                        mkdir('-p', path.join(taskPath, 'node_modules'));
-                        rm('-Rf', path.join(taskPath, 'node_modules', modName));
-                        pushd(taskPath);
-                        run('npm install ' + modOutDir);
-                        popd();
+            taskMake = require(makePath);
+
+            // external .zip files
+            if (taskMake.hasOwnProperty('archivePackages')) {
+                var archivePackages = taskMake.archivePackages;
+                archivePackages.forEach(function (archive) {
+                    assert(archive.url, 'archive.url');
+                    assert(archive.dest, 'archive.dest');
+
+                    // download and extract the archive package
+                    var archiveSource = downloadArchive(archive.url);
+
+                    // copy the files
+                    var archiveDest = path.join(outDir, archive.dest);
+                    mkdir('-p', archiveDest);
+                    cp('-R', archiveSource, archiveDest)
+                });
+            }
+
+            // external NuGet V2 packages
+            if (taskMake.hasOwnProperty('nugetv2')) {
+                var nugetPackages = taskMake.nugetv2;
+                nugetPackages.forEach(function (package) {
+                    // validate the structure of the json data
+                    assert(package.name, 'package.name');
+                    assert(package.version, 'package.version');
+                    assert(package.repository, 'package.repository');
+                    assert(package.cp, 'package.cp');
+                    assert(package.cp, 'package.cp.length');
+
+                    // download and extract the NuGet V2 package
+                    var url = package.repository.replace(/\/$/, '') + '/package/' + package.name + '/' + package.version;
+                    var packageSource = downloadArchive(url, /*omitExtensionCheck*/true);
+
+                    // copy specific files
+                    copyGroups(package.cp, packageSource, outDir);
+                });
+            }
+
+            //--------------------------------
+            // Common: build, copy, install 
+            //--------------------------------
```

Comment:
i got confused and moved all of this stuff inside the `if (test('-f', makePath)) {` block. i realize now it was technically fine before, i can move it back out so the diff isnt so hard to follow.